src/components/Widgets/RelationControl.js

```diff
@@ -31,36 +25,50 @@ class RelationControl extends Component {
     super(props, ctx);
     this.controlID = uuid.v4();
     this.didInitialSearch = false;
+    this.state = { displayValue: '' };
   }
 
   componentDidMount() {
     const { value, field } = this.props;
     if (value) {
       const collection = field.get('collection');
-      const searchFields = field.get('searchFields').toJS();
+      const searchFields = [...new Set(
+        [
+          ...field.get('searchFields').toJS(),
+          field.get('valueField'),
```

Comment:
Can you explain the decision to move to a `Set` and to pass in the value field?

Reply:
The move to a set (casted to an array) is just a way to ensure unique values in the array, since `searchFields` could contain `valueField`.

The actual initial search (prior to my mods) is to find a search match on the stored value.
Is its purpose to preserve relations when the target field has changed a bit ?

i.e. I change my `page` collection item title from "Job offers" to "Job listings".
A `menu` collection item that has a `relation` set to the "Job offers" page title field has a now a broken relation.
When i go to edit the `menu`, the initial search is performed and let's say an unique match is made on "Job listings" (probably because it's the only page containing the word "Job"). The relation value field will be automatically set to "Job Listings".
This behavior is not really ideal, since in this case the relation entry shows an automagically fixed relation, that is not even correct until saved, and it is really misleading.
I hope i make sense on this. 

I think this behavior could be changed to only perform the search in `valueField`, and get the exact match needed to get the `displayValue`.
I really think that the main use cases of relations should be implemented by linking to a unique constant field per item, or face broken relations, and that broken relations should not be automatically corrected.

This code was just a mix of both approaches, getting the `displayValue` and preserving the old behavior.
Do you think we should keep the actual behavior ?
Or that i should change this code to
```
const searchFields =  [ field.get('valueField') ];
```
to only get the initial `displayValue` and be done with it ?